trust-center-agent/src/test/java/care/smith/fts/tca/deidentification/FhirPseudonymProviderTest.java

```diff
@@ -73,7 +76,25 @@ void setUp(MockServerClient mockServer) {
 
   @Test
   void retrieveTransportIds(MockServerClient mockServer) throws IOException {
-    var fhirGenerator = FhirGenerators.gpasGetOrCreateResponse(() -> "id1", () -> "469680023");
+    var idIterator = List.of("id1", "Salt_id1").iterator();
+    var valueIterator = List.of("469680023", "123").iterator();
+
+    var fhirGenerator =
+        FhirGenerators.gpasGetOrCreateResponse(
+            () -> {
+              if (idIterator.hasNext()) {
+                return idIterator.next();
+              } else {
+                return null;
+              }
+            },
+            () -> {
+              if (valueIterator.hasNext()) {
+                return valueIterator.next();
+              } else {
+                return null;
+              }
+            });
```

Comment:
This looks awful. Can we use `TestPatientGenerator.fromList` instead and move it to `FhirGenerators`?